src/tox/_pytestplugin.py

```diff
@@ -210,36 +210,47 @@ def wait(self):
 
 @pytest.fixture(name="mocksession")
 def create_mocksession(request):
-    class MockSession(Session):
-        def __init__(self):
-            self._clearmocks()
-            self.config = request.getfixturevalue("newconfig")([], "")
-            self.resultlog = ResultLog()
-            self._actions = []
+    config = request.getfixturevalue("newconfig")([], "")
 
-        def getenv(self, name):
-            return VirtualEnv(self.config.envconfigs[name], session=self)
-
-        def _clearmocks(self):
+    class MockSession(Session):
+        def __init__(self, config):
+            self.logging_levels(config.option.quiet_level, config.option.verbose_level)
+            super(MockSession, self).__init__(config, popen=self.popen)
             self._pcalls = []
-            self._spec2pkg = {}
-            self.report = ReportExpectMock(self)
+            self.report = ReportExpectMock()
 
-        def make_emptydir(self, path):
-            pass
+        def _clearmocks(self):
+            if six.PY3:
+                self._pcalls.clear()
+            else:
+                del self._pcalls[:]
+            self.report.clear()
 
         def popen(self, args, cwd, shell=None, stdout=None, stderr=None, env=None, **_):
-            pm = pcallMock(args, cwd, env, stdout, stderr, shell)
-            self._pcalls.append(pm)
-            return pm
+            process_call_mock = pcallMock(args, cwd, env, stdout, stderr, shell)
+            self._pcalls.append(process_call_mock)
+            return process_call_mock
+
+        def new_config(self, config):
+            self.logging_levels(config.option.quiet_level, config.option.verbose_level)
+            self.config = config
+            self.venv_dict.clear()
+            self.existing_venvs.clear()
+
+        def logging_levels(self, quiet, verbose):
+            update_default_reporter(quiet, verbose)
+            if hasattr(self, "config"):
+                self.config.option.quiet_level = quiet
+                self.config.option.verbose_level = verbose
 
-    return MockSession()
+    return MockSession(config)
 
 
 @pytest.fixture
 def newmocksession(mocksession, newconfig):
     def newmocksession_(args, source, plugins=()):
-        mocksession.config = newconfig(args, source, plugins=plugins)
+        config = newconfig(args, source, plugins=plugins)
+        mocksession.__init__(config)
```

Comment:
interesting, I've never seen this before -- I assume it just re-initializes the object? is this well defined?

Reply:
It's just a function call AFAIK 👍

Reply:
seems [well defined but ill-advised](https://stackoverflow.com/questions/13091221/reinitialize-an-object-with-self-init)

Reply:
because we want to reset the report, calculated session venvs when a new config is loaded 👍 can use the _reset path though.